File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/storage/ingrediente/IngredienteStorageServiceSerializable.java
package storage.ingrediente;

import config.ConfigApp;
import models.Ingrediente;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class IngredienteStorageServiceSerializable implements IngredienteStorageService {

    ConfigApp configApp = ConfigApp.getInstance();

    File file = new File(configApp.getAPP_DATA()+File.separator+"ingrediente.ser");

    public IngredienteStorageServiceSerializable() throws IOException {}

    @Override
    public void saveAll(List<Ingrediente> entities) throws IOException {
        try(ObjectOutputStream objectOutputStream = new ObjectOutputStream(new FileOutputStream(file))){
            objectOutputStream.writeObject(entities);
        }
    }

    @Override
    public List<Ingrediente> loadAll() throws IOException {
        List<Ingrediente> ingredientes = new ArrayList<>();
        if(!file.exists()) return ingredientes;
        try(ObjectInputStream objectInputStream = new ObjectInputStream(new FileInputStream(file))){
            ingredientes = (List<Ingrediente>) objectInputStream.readObject();
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        }
        return ingredientes;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/storageService/Ingrediente/IngredienteStorageServiceSerializable.kt
package storageService.Ingrediente

import config.ConfigApp
import model.Ingrediente
import mu.KotlinLogging
import java.io.*

class IngredienteStorageServiceSerializable: IngredienteStorageService {

    private val config = ConfigApp
    private val file = File(config.APP_DATA+ File.separator+"ingredientes.ser")
    private val logger = KotlinLogging.logger {}

    override fun saveAll(entities: List<Ingrediente>) {
        logger.debug { "Se guardan todos los ingredientes en el fichero serializable." }
        ObjectOutputStream(FileOutputStream(file)).use{
            it.writeObject(entities)
        }
    }

    override fun loadAll(): List<Ingrediente> {
        logger.debug { "Se cargán todos los ingredientes del fichero serializable." }
        var ingredientes = mutableListOf<Ingrediente>()
        if(!file.exists()) return ingredientes.toList()
        ObjectInputStream(FileInputStream(file)).use{
            ingredientes = it.readObject() as MutableList<Ingrediente>
        }
        return ingredientes.toList()
    }
}